// firebase-tools/lib/mcp/tools/dataconnect/get_dataconnect_schema.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.get_dataconnect_schema = void 0;
const zod_1 = require("zod");
const tool_js_1 = require("../../tool.js");
const util_js_1 = require("../../util.js");
const client = require("../../../dataconnect/client");
const fileUtils_js_1 = require("../../../dataconnect/fileUtils.js");
const converter_js_1 = require("./converter.js");
exports.get_dataconnect_schema = (0, tool_js_1.tool)({
    name: "get_dataconnect_schema",
    description: "List the Firebase Data Connect Schema in the project, which includes Cloud SQL data sources and the GraphQL Schema describing what tables are available.",
    inputSchema: zod_1.z.object({
        serviceId: zod_1.z
            .string()
            .nullable()
            .describe("The Firebase Data Connect service ID to look for. By default, it would pick the the service ID project directory."),
    }),
    annotations: {
        title: "Obtain the Firebase Data Connect Schemas that's available in the backend",
        readOnlyHint: true,
    },
    _meta: {
        requiresProject: true,
        requiresAuth: true,
    },
}, async ({ serviceId }, { projectId, config }) => {
    const serviceInfo = await (0, fileUtils_js_1.pickService)(projectId, config, serviceId || undefined);
    const schemas = await client.listSchemas(serviceInfo.serviceName, ["*"]);
    return (0, util_js_1.toContent)(schemas === null || schemas === void 0 ? void 0 : schemas.map(converter_js_1.schemaToText).join("\n\n"));
});
